doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -0,0 +1,190 @@
+    ... 
+    >>> dist = StandardNormal()
+    >>> 
+    >>> urng = np.random.default_rng()
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+    >>> rng.rvs()
+    -1.526829048388144
+
+In the above example, we have used the TDR method to sample from the standard
+normal distribution. Note that we can drop the normalization constant while
+computing the PDF. This usually helps speed up the sampling stage. Also, note
+that the PDF doesn't need to be vectorized. It should accept and return a
+scalar.
+
+It is also possible to evaluate the inverse of the CDF of the hat distribution
+directly using the ``ppf_hat`` method.
+
+    >>> rng.ppf_hat(0.5)
+    -0.00018050266342362759
+    >>> norm.ppf(0.5)
+    0.0
+    >>> u = np.linspace(0, 1, num=10)
+    >>> rng.ppf_hat(u)
+    array([       -inf, -1.22227372, -0.7656556 , -0.43135731, -0.14002921,
+            0.13966423,  0.43096141,  0.76517113,  1.22185606,         inf])
+    >>> norm.ppf(u)
+    array([       -inf, -1.22064035, -0.76470967, -0.4307273 , -0.1397103 ,
+            0.1397103 ,  0.4307273 ,  0.76470967,  1.22064035,         inf])
+
+Apart from the PPF method, other attributes can be accessed
+to see how well the generator fits the given distribution. These are:
+
+* 'sqhratio': (area below squeeze) / (area below hat) for the generator. It
+  is a number between 0 and 1. Closer to 1 means that the hat and the squeeze
+  functions tightly envelop the distribution and fewer PDF evaluations are
+  required to generate samples. The expected number of evaluations of the
+  density is bounded by ``(1/sqhratio) - 1`` per sample. By default, it is
+  kept above 0.99 but that can be changed by passing a ``max_sqhratio``
+  parameter.
+* 'hat_area': area below the hat for the generator.
+* 'squeeze_area': area below the squeeze for the generator.
+
+    >>> rng.sqhratio
+    0.9947024204884917
+    >>> rng.hat_area
+    2.510253139791547
+    >>> rng.squeeze_area
+    2.4969548741894876
+    >>> rng.sqhratio == rng.squeeze_area / rng.hat_area
+    True
+
+To increase ``sqhratio``, pass ``max_sqhratio``:
+
+    >>> rng = TransformedDensityRejection(dist, max_sqhratio=0.999,
+    ...                                   max_intervals=1000, seed=urng)
+    >>> rng.sqhratio
+    0.999364900465214
+
+Note that we need to increase the ``max_intervals`` parameter when we want
+a higher ``sqhratio``. This is because more construction points are required
+to fit the distribution more tightly.
+
+Let's see how this affects the callbacks to the PDF method of the
+distribution:
+
+    >>> class StandardNormal:
+    ...     def __init__(self):
+    ...         self.callbacks = 0
+    ...     def pdf(self, x):
+    ...         self.callbacks += 1
+    ...         return np.exp(-0.5 * x*x)
+    ...     def dpdf(self, x):
+    ...         return -x * np.exp(-0.5 * x*x)
+    ... 
+    >>> dist1 = StandardNormal()
+    >>> urng = np.random.default_rng(0x7344aca86f1dafd8820be998b639551c)
```

Comment:
Here too

Reply:
We need to fix the seed here to demonstrate the callbacks made by UNU.RAN when two different values of a parameter are passed for the same stream of random numbers. As the seed is quire arbitrary, I figured this would be OK.

Reply:
If we must then fine. An alternative is to do a copy.

Reply:
> An alternative is to do a copy.

Good idea, thanks. Let me try that.

Reply:
Thanks, IMHO it's better.

Reply:
@tupui I have changed this according to your suggestion